core/src/main/java/org/frankframework/receivers/Receiver.java

```diff
@@ -1191,7 +1211,10 @@ private Message processMessageInAdapter(MessageWrapper<M> messageWrapper, PipeLi
 				return Message.nullMessage();
 			}
 
-			IbisTransaction itx = new IbisTransaction(txManager, getTxDef(), "receiver [" + getName() + "]");
+			// If the receiver is transacted, there should already be a transaction. We want to hook in to this existing
+			// transaction, so we can query the status, but not create a new one if the transceiver shouldn't be transacted.
+			// Therefore, we use here PROPAGATION_SUPPORTS
+			IbisTransaction itx = new IbisTransaction(txManager, TXSUPPORTED, "receiver [" + getName() + "]");
```

Comment:
@evandongen This is the change that ensures the receiver process messages in _existing_ transaction, instead of potentially creating a new transaction.

Other changes in Receiver are mostly to make sure that the stated precondition of the methods `processMessageInAdapter()` and `processRawMessage()` is met: that a transaction already exists, if the receiver is supposed to be transacted.

So there are some changes to create a transaction before calling one of these methods, in `retryMessage()` and `processRequest()`.